apps/omg_eth/lib/omg_eth/client.ex

```diff
@@ -29,13 +29,16 @@ defmodule OMG.Eth.Client do
     end
   end
 
-  @spec node_ready() :: :ok | {:error, :geth_still_syncing | :geth_not_listening}
-  @spec node_ready(module()) :: :ok | {:error, :geth_still_syncing | :geth_not_listening}
+  @spec node_ready() :: :ok | Ethereumex.Client.Behaviour.error()
+  @spec node_ready(module()) :: :ok | Ethereumex.Client.Behaviour.error()
   def node_ready(client \\ Ethereumex.HttpClient) do
     case client.eth_syncing() do
       {:ok, false} -> :ok
       {:ok, _} -> {:error, :geth_still_syncing}
       {:error, :econnrefused} -> {:error, :geth_not_listening}
+
+      other ->
+        other
```

Comment:
```suggestion
      {:error, :closed} = other->
        other
```

Reply:
Do we want to strictly match it for {:error, :closed} or should we be open to all the other errors

Reply:
let's match on **what we know happens occasionally** and **leave** to _fail_ for responses we don't know